theano/scalar/basic.py

```diff
@@ -1513,10 +1528,13 @@ def impl(self, *inputs):
 
     def c_code(self, node, name, inputs, outputs, sub):
         (z,) = outputs
+        op = " * "
+        if z.type == bool:
```

Comment:
Here, `z` is the name of the variable in the C code, it's a string.
You probably want `node.outputs[0].type.dtype` or something like that.